include/dyad/stream/dyad_stream_api.hpp

```diff
@@ -88,8 +95,103 @@ void fsync_fstream (std::basic_fstream<_CharT, _Traits>& os)
         os.flush ();
         fsync (static_cast<my_filebuf&> (*os.rdbuf ()).handle ());
     }
+#else
+    if (os.is_open ()) {
+        os.flush ();
+    }
+#endif // DYAD_HAS_STD_FSTREAM_FD
+}
+
+//----------------------------------------------------------------------
+#if defined(DYAD_HAS_STD_FSTREAM_FD)
+//----------------------------------------------------------------------
+template <typename _CharT, typename _Traits>
+void lock_ofstream (std::basic_ofstream<_CharT, _Traits>& os, const dyad_stream_core& core)
+{
+    class my_filebuf : public std::basic_filebuf<_CharT>
+    {
+       public:
+        int handle ()
+        {
+            return this->_M_file.fd ();
+        }
+    };
+
+    if (os.is_open ()) {
+        int fd = static_cast<my_filebuf&> (*os.rdbuf ()).handle ();
+        core.file_lock (fd);
```

Comment:
Is there a way to select which lock your applying? We should use shared locks for read and exclusive locks for writes,

Reply:
ah i see u are only only applying for producer. I think, we should have different methods for shared or exclusive locks for clarity of API.

Reply:
I know maybe u wanna say exclusive lock

Reply:
We can add a shared lock to ifstream. Also to the fstream with `m_core.is_dyad_consumer()`

Reply:
We don't need that as the consume code already does shared lock.

Reply:
I agree. That is why I  did not add it.